source/rust_verify/src/verifier.rs

```diff
@@ -591,16 +683,48 @@ impl Verifier {
                 let error_as = if recommends_rerun { ErrorAs::Note } else { ErrorAs::Error };
                 let s =
                     if recommends_rerun { "Function-Check-Recommends " } else { "Function-Def " };
-                let invalidity = self.run_commands_queries(
-                    compiler,
-                    error_as,
-                    air_context,
-                    &commands,
-                    &HashMap::new(),
-                    &snap_map,
-                    &(s.to_string() + &fun_as_rust_dbg(&function.x.name)),
-                );
-                if invalidity && !recommends_rerun && !self.args.no_auto_recommends_check {
+
+                let mut function_invalidity = false;
+                for command in commands.iter().map(|x| &*x) {
+                    let CommandsWithContextX { span, desc: _, commands: _, spinoff_z3 } =
+                        &**command;
+                    let command_invalidity: bool;
+                    if *spinoff_z3 {
+                        let mut spinoff_air_context = self.new_air_context_with_module_context(
+                            ctx,
+                            module,
+                            &(function.x.name).path,
+                            datatype_commands.to_vec(),
+                            function_decl_commands.to_vec(),
+                            function_spec_commands.to_vec(),
+                            function_axiom_commands.to_vec(),
```

Comment:
This copies all the vectors for every `self.new_air_context_with_module_context` call, which doesn't seem ideal. Can we pass a reference / clone a wrapper `Arc` instead?

Reply:
Yes, that sounds good. Thanks!